File: nb_cli_plugin_webui/app/status/schemas.py
from typing import List, Optional

from pydantic import BaseModel

from nb_cli_plugin_webui.app.handlers.process import ProcessInfo


class PlatformProfile(BaseModel):
    name: str
    struct: str
    platform_type: str


class CpuInfo(BaseModel):
    name: str
    count: int
    max_freq: str
    current_freq: str
    percent: float
    process: int


class MemInfo(BaseModel):
    total: int
    available: int
    percent: float
    used: int
    free: int


class DiskInfo(BaseModel):
    total: int
    used: int
    free: int
    speed: List[int]


class NetInfo(BaseModel):
    sent_total: int
    recv_total: int
    package_sent: int
    package_recv: int
    speed: List[int]


class SystemInfo(BaseModel):
    platform: PlatformProfile
    cpu: CpuInfo
    mem: MemInfo
    disk: DiskInfo
    net: NetInfo


class StatusInfo(BaseModel):
    system: SystemInfo
    process: Optional[ProcessInfo]

File: frontend/src/components/HomePage/schemas.ts
import type { ProcessInfo } from "@/api/schemas";

interface PlatformProfile {
  name: string;
  struct: string;
  platform_type: string;
}

interface Cpu {
  name: string;
  count: number;
  max_freq: string;
  current_freq: string;
  percent: number;
  process: number;
}

interface Mem {
  total: number;
  available: number;
  percent: number;
  used: number;
  free: number;
}

interface Disk {
  total: number;
  used: number;
  free: number;
  speed: number[];
}

interface Net {
  sent_total: number;
  recv_total: number;
  package_sent: number;
  package_recv: number;
  speed: number[];
}

interface SystemInfo {
  platform: PlatformProfile;
  cpu: Cpu;
  mem: Mem;
  disk: Disk;
  net: Net;
}

export interface StatusInfo {
  system: SystemInfo;
  process?: ProcessInfo;
}
